### PR TITLE
Support parquet file format and add batch and filter_by options

### PR DESCRIPTION
**Pull request by benoit-cty**
#### New features

- Support parquet file format : 
    - if a parquet file is provided to buildcollection it won't be converted in HDF5 but added to the collection as is.
    - It is possible to provide a folder containing many files : each files will be used as a separate table.
 
- Run a simulation on a part of the input datasets - for the moment it works only for two entities simulations :
    - With a batch size option
    - With a filter_by option 

### Sample collection

This is how a parquet collection looks like:

```json
{
  "name": "test_parquet_collection",
  "surveys": {
    "test_parquet_collection_2020": {
      "hdf5_file_path": null,
      "parquet_file_path": "/media/2To-nvme/dev/src/LEXIMPACT/openfisca-survey-manager/openfisca_survey_manager/tests/data_files/test_parquet_collection",
      "label": "test_parquet_collection_2020",
      "name": "test_parquet_collection_2020",
      "tables": {
        "household": {
          "source_format": "parquet",
          "variables": [
            "household_id",
            "rent",
            "household_weight"
          ],
          "parquet_file": "/media/2To-nvme/dev/src/LEXIMPACT/openfisca-survey-manager/openfisca_survey_manager/tests/data_files/test_parquet_collection/household.parquet"
        },
        "person": {
          "source_format": "parquet",
          "variables": [
            "household_id",
            "salary",
            "person_weight",
            "household_role_index"
          ],
          "parquet_file": "/media/2To-nvme/dev/src/LEXIMPACT/openfisca-survey-manager/openfisca_survey_manager/tests/data_files/test_parquet_collection/person.parquet"
        }
      },
      "informations": {
        "csv_files": [],
        "parquet_files": [
          "/media/2To-nvme/dev/src/LEXIMPACT/openfisca-survey-manager/openfisca_survey_manager/tests/data_files/test_parquet_collection/household.parquet",
          "/media/2To-nvme/dev/src/LEXIMPACT/openfisca-survey-manager/openfisca_survey_manager/tests/data_files/test_parquet_collection/person.parquet"
        ],
        "sas_files": [],
        "stata_files": []
      }
    }
  }
}
```

### CODE REVIEW

File: openfisca_survey_manager/survey_collections.py [hunk 1]
```diff
@@ -72,14 +76,14 @@ def dump(self, config_files_directory = None, json_file_path = None):
             self.json_file_path = json_file_path
 
         config.set("collections", self.name, self.json_file_path)
-        config.save
+        config.save  # TODO: check if this is necessary : why two call to save ?
```
Comment (benjello) [inline openfisca_survey_manager/survey_collections.py:79]: I think you candefinitely remobe this line

File: openfisca_survey_manager/scripts/build_collection.py [hunk 2]
```diff
@@ -134,12 +144,19 @@ def build_survey_collection(
             os.mkdir(collections_directory)
         collection_json_path = os.path.join(collections_directory, "{}.json".format(collection_name))
         survey_collection.dump(json_file_path = collection_json_path)
-        surveys = [survey for survey in survey_collection.surveys if survey.name.endswith(str(survey_suffix))]
+        surveys = []
+        for survey in survey_collection.surveys:
+            # TODO: Check with Mahdi why all suveys of the same year was added to the collection and not only the one we provide in parameters ?
```
Comment (benoit-cty) [inline openfisca_survey_manager/scripts/build_collection.py:149]: @benjello Do you remember why ?

Comment (benjello) [reply openfisca_survey_manager/scripts/build_collection.py:149]: Which parameters are you talking about ?
Aren't they in `data_directory_path_by_survey_suffix` ?

Comment (benoit-cty) [reply openfisca_survey_manager/scripts/build_collection.py:149]: I mean, in `def build_collection(` we provide a `collection_name` that is used to build the `survey_name` so why the original loop is rebuiding all the surveys in the collection for the year, as `survey_suffix` contains the year.

Comment (benjello) [reply openfisca_survey_manager/scripts/build_collection.py:149]: I don't remember @benoit-cty

**Comment by benoit-cty** [comment]
Tested with OpenFisca-France-Data and [the full CI works](https://git.leximpact.dev/benjello/openfisca-france-data/-/pipelines/12383).

File: openfisca_survey_manager/tests/data_files/test_random_generator.json
Comment (benoit-cty) [inline openfisca_survey_manager/tests/data_files/test_random_generator.json:1]: This file is generated by the CI.

File: setup.py [hunk 3]
```diff
@@ -89,6 +89,7 @@
         'humanize >=4.6.0, < 5.0',
         'openfisca-core >=41.1.0, < 42.0',
         'pandas >=2.0.3, < 3.0',
+        'pyarrow >=13.0.0, < 14.0.0',
```
Comment (benoit-cty) [inline setup.py:92]: To be able to read parquet files in batch.

**Comment by clallemand** [comment]
@benoit-cty maybe  we could wrap up this work? I have made some modifications by testing the code on the data on the CASD

**Comment by benjello** [comment]
Le ‘parquet_file_path‘ est plutôt un ‘parquet_dir_path‘ non ?

**Comment by clallemand** [comment]
@benjello, @sylvainipp, @Lolajossipp : this PR add new features we have developed to run simulations with administrative fiscal data (POTE). It works with the data, we did simulations on the CASD. Don't hesitate if you have any questions, it could be difficult to review.
For the moment the new features work only if you have a two entities simulation but it doesn't break old features. So it shouldn't change things for your simulations.

**Comment by clallemand** [comment]
> Le ‘parquet_file_path‘ est plutôt un ‘parquet_dir_path‘ non ?

It could be weither a file or a dir path. Both work

**Review by benjello — APPROVED** [review]

**Comment by sylvainipp** [comment]
Thanks for this new feature ! It will be really helpful for the next update of our data ! I wanted to suggest typo changes but I made a mistake and directly committed, sorry for that.

**Comment by clallemand** [comment]
> Thanks for this new feature ! It will be really helpful for the next update of our data ! I wanted to suggest typo changes but I made a mistake and directly committed, sorry for that.

Thanks @sylvainipp ! Can I merge ?

**Review by sylvainipp — APPROVED** [review]

**Pull request merged**